datafusion/expr/src/expr_rewriter/order_by.rs

```diff
@@ -116,7 +116,19 @@ fn rewrite_in_terms_of_projection(
 
         // look for the column named the same as this expr
         if let Some(found) = proj_exprs.iter().find(|a| expr_match(&search_col, a)) {
```

Comment:
`into_iter` might prevent the clone later on?

Reply:
`into_iter` would consume `proj_exprs` which can not be done here because it is captured in closure which is passed to `rewrite_expr`

Reply:
ok : 👌